src/spring-mongo/src/main/java/net/conology/restjsonpath/JsonPathCompilerPass.java

```diff
@@ -42,12 +57,12 @@ private PropertyFilterNode transform(JsonPathMongoParser.RestAndQueryContext ctx
         }
 
         var expressions = ctx.restBasicQuery()
-            .stream()
-            .map(this::transform)
-            .toList();
+                .stream()
+                .map(this::transform)
+                .toList();
 
         return expressions.size() == 1 ? expressions.getFirst()
-            : new AndFilterNode(expressions);
+                : new AndFilterNode(expressions);
```

Comment:
Maybe it would be easier to always use these nodes, even if singleton. We should keep an eye on that